### PR TITLE
Voting Weight

### PR DESCRIPTION
**Pull request by Raemon**
This PR makes it so that
a) when you click on a vote button on the /reviewVoting page, it updates the vote-cost-total
b) when your vote-cost-total is over 500, your vote-weights are reduced.

It does not yet make it so that the vote-weights are visibly reduced in other parts of the site. 

It currently accomplishes A) via changing currentUserReviewVote into an object that includes the _id of the reviewVote, so that updating the reviewVote updates the the datastructure. But, it seems kinda sluggish and inconsistent and I'm not sure this is actually better than the previous solution of "you just have to refresh the page."

It accomplishes b) via passing down the costTotal of the votes and using that to do math to the point-values. 

**Some underlying philosophy:** Initially, I was going to solve the "what about people with more than 500-vote-cost?" by saying "you just can't spend more than 500 points. If you try, it warns you and disables buttons until you reduce your cost." But then, as a user, I found it fairly important to actually be able to differentiate my 9s from 4s where appropriate. The "500 max" rule made it so that I had to artificially lower my scores for a lot of posts that I thought were quite good.

I think it's important people can just vote on everything that feels qualitatively-very-important to them. 

image

### CODE REVIEW

File: packages/lesswrong/components/review/ReviewVotingButtons.tsx [hunk 1]
```diff
@@ -51,59 +53,74 @@ const styles = (theme: ThemeType) => ({
   7: { color: upvoteColor},
 })
 
-export const indexToTermsLookup = {
-  0: {label: null, cost: 0, tooltip: null},
-  1: { label: "-9", cost: 45, tooltip: 
+const getPointsFromCost = (cost) => {
+  // the formula to quadratic cost from a number of points is (n^2 + n)/2
+  // this uses the inverse of that formula to take in a cost and output a number of points
+  return (-1 + Math.sqrt((8 * cost)+1)) / 2
```
Comment (Raemon) [inline packages/lesswrong/components/review/ReviewVotingButtons.tsx:59]: @Discordius check the math here?

**Comment by Raemon** [comment]
(latest commits aren't working properly. I'd appreciate @Discordius giving a look if he has time)

File: packages/lesswrong/components/review/ReviewVotingButtons.tsx [hunk 2]
```diff
@@ -51,59 +52,23 @@ const styles = (theme: ThemeType) => ({
   7: { color: upvoteColor},
 })
 
-export const indexToTermsLookup = {
```
Comment (Raemon) [inline packages/lesswrong/components/review/ReviewVotingButtons.tsx:54]: I moved this over to reviewUtils to avoid a circular dependency lint error

**Comment by Raemon** [comment]
@jpaddison3 I think it now makes sense for you to review this in depth. Oli took a detailed look at the PR to fix various responsivenes issues and UI jankiness. Some notes:

– to get the ReviewVotingPage to update quickly when you vote (i.e. showing an updated total and updating the weights of your votes), we switched the submitReviewVote mutation from returning a reviewVote, to returning a post (which contains a review vote). This propagates faster and reliably through the data tree.

– the PostsPage review buttons unfortunately don't display the updated weightings. Getting that to work properly is more work than I have time for this week.

– I've gone ahead and removed the various branches of the tree that involve "useQuadratic". I think the new voting system is flexible, we can tweak the UI next year to make it more "quadratic-voting-like" if we want without having two entire voting paths

**Comment by Raemon** [comment]
Things to test:
[] creating a new vote on the /reviewVoting page. (verify it updates on /reviewVoting, and also on the corresponding post page)
[] updating a new vote on the /reviewVoting page. (verify it updates on /reviewVoting as well as 

[]  Verify vote tooltip displays properly when you've spent less than 500 points
[]  Verify vote tooltip displays properly when you've spent  more than 500 points
[] Verify vote-cost-total tooltip displays properly when

File: packages/lesswrong/lib/reviewUtils.tsx [hunk 3]
```diff
@@ -67,3 +69,64 @@ export const currentUserCanVote = (currentUser: UsersCurrent|null) => {
...
+  // this uses the inverse of that formula to take in a cost and output a number of points
+  return (-1 + Math.sqrt((8 * cost)+1)) / 2
+}
+
+const getLabelFromCost = (cost) => {
+  // rounds the points to 1 decimal for easier reading
+  return round(getPointsFromCost(cost), 1)
+}
+
+export const getCostData = ({costTotal=500}:{costTotal?:number}) => {
```
Comment (Discordius) [inline packages/lesswrong/lib/reviewUtils.tsx:84]: Why does this take an object-argument? I mean, maybe fine for future-proofness, but this feels like it should just take a single argument that's a number.

**Review by Discordius — APPROVED** [review]

File: packages/lesswrong/lib/reviewUtils.tsx (hunk 3 above)
Comment (Raemon) [reply packages/lesswrong/lib/reviewUtils.tsx:84]: I thought I had to do that to give it a default value. Not sure if that was necessary.

**Pull request merged**